.github/workflows/integrate.yaml

```diff
@@ -49,77 +49,88 @@ jobs:
     - name: Check out repo
       uses: actions/checkout@v3
 
-    - uses: balchua/microk8s-actions@v0.3.2
+    - name: Setup operator environment
+      uses: charmed-kubernetes/actions-operator@main
       with:
-        # TODO: Pinned to <1.25 until we update to istio 1.16
-        channel: '1.24/stable'
-        addons: '["dns", "storage", "rbac", "metallb:10.64.140.43-10.64.140.49"]'
-
-    - name: Install dependencies
-      run: |
-        set -eux
-        sudo snap install charmcraft --classic --channel=latest/candidate
-        sudo snap install juju --classic
-        sudo snap install juju-bundle --classic
-        sudo snap install juju-wait --classic
-        sudo apt update
-        sudo apt install tox
-
-    - name: Wait before bootstrap
+        provider: microk8s
+        channel: 1.24/stable
+        charmcraft-channel: latest/candidate
+        # TODO: Unpin this when this bug is resolved: https://bugs.launchpad.net/juju/+bug/1992833.
+        #       In particular, these tests failed deploying the prometheus-k8s charm where it gets an error in
+        #       the "metrics-endpoint-relation-changed" hook.
+        bootstrap-options: --agent-version="2.9.34"
+        microk8s-addons: "dns storage rbac metallb:10.64.140.43-10.64.140.49"
+    # TODO: Remove once the actions-operator does this automatically
+    - name: Configure kubectl
       run: |
-        set -eux
-        sg microk8s -c 'microk8s status --wait-ready'
-        kubectl wait --for=condition=available --timeout=5m -nkube-system deployment/coredns deployment/hostpath-provisioner
+        sg microk8s -c "microk8s config > ~/.kube/config"
 
-    - name: Bootstrap
+    - name: Add model
       run: |
-        # Pinning juju agent to 2.9.42 to keep compatibility with pythonlib-juju<3
-        sg microk8s -c 'juju bootstrap microk8s uk8s --agent-version=2.9.42'
-        sg microk8s -c 'juju add-model kubeflow'
+        juju add-model kubeflow
+        juju switch kubeflow
 
+    # DEBUG: added keep-models to ensure our logs are available during dumping
     - run: sg microk8s -c 'KUBECONFIG=/home/runner/.kube/config tox -e integration
-        -- --model kubeflow --destructive-mode'
+        -- --model kubeflow --destructive-mode --keep-models'
       timeout-minutes: 80
 
     - name: Setup Debug Artifact Collection
       run: mkdir tmp
-      if: failure()
+      if: always()
```

Comment:
Let's use your action?

Reply:
Yeah can do.  I'll need to put it somewhere in our org first